src/lib/utils/letterDataUtils.test.ts

```diff
@@ -93,3 +101,173 @@ describe('makeWordUsed', () => {
 		});
 	});
 });
+
+describe('provideHint', () => {
+	it('should populate a hint into playedLetters', () => {
+		let playedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'hint' },
+			{ value: 'o', state: 'inPlay' }
```

Comment:
shouldn't this `'o'` be in `unplayedLetters`?

Reply:
I want the test to account for the user hitting the hint button while they are in the middle of solving a word. Its like if the user typed 'o', then went to push the hint button.